.github/workflows/mingw.yml

```diff
@@ -0,0 +1,225 @@
+          export ac_cv_have_decl_RTLD_NOLOAD=no
+          
+          export CFLAGS
+          export CXXFLAGS
+          export CPPFLAGS
+          
+          
+          MSYSTEM=MINGW ./configure \
+            --prefix=${MINGW_PREFIX} \
+            --host=${MINGW_CHOST} \
+            --build=${MINGW_CHOST} \
+            --enable-shared \
+            --with-nt-threads \
+            --with-system-expat \
+            --with-system-ffi \
+            --with-system-libmpdec \
+            --without-ensurepip \
+            --without-c-locale-coercion \
+            --enable-loadable-sqlite-extensions \
+            "${_extra_config[@]}" \
+            RCFLAGS="$rc_flags" \
+            OPT=""
+          # We patch importlib which is embedded in C headers, so regenerate them
+          make regen-importlib
+
+          make -j8
+          
+          # Add missing venvlauncher files (issue #7014)
+          # FIXME: build these from PC/launcher.c instead
+          cp python.exe venvlauncher.exe
+          cp pythonw.exe venvwlauncher.exe
+
+      - name: Install
+        shell: msys2 {0}
+        run: |
+          set -ex
+          
+          pkgver='3.9.5'
+          _pybasever='3.9'
+          srcdir=.
+
+          # now install things
+
+          pkgdir=python_pkgdir
+
+          PREFIX_WIN=$(cygpath -wm ${MINGW_PREFIX})
+          MSYSTEM=MINGW \
+          MSYS2_ARG_CONV_EXCL="--prefix=;--install-scripts=;--install-platlib=" \
+          make -j1 install DESTDIR="${pkgdir}"
+          VERABI=${_pybasever}
+
+          # gdb pretty printers for debugging Python itself; to use:
+          # python
+          # sys.path.append('C:/msys64/mingw64/share/gdb/python3')
+          # import python_gdb
+          # reload(python_gdb)
+          # end
+
+          cp -f "${pkgdir}${MINGW_PREFIX}"/lib/python${_pybasever}/config-${VERABI}/libpython${VERABI}.dll.a "${pkgdir}${MINGW_PREFIX}"/lib/libpython${VERABI}.dll.a
+
+          # Need for building boost python module
+          cp -f "${pkgdir}${MINGW_PREFIX}"/lib/python${_pybasever}/config-${VERABI}/libpython${VERABI}.dll.a "${pkgdir}${MINGW_PREFIX}"/lib/libpython${_pybasever}.dll.a
+
+          for fscripts in 2to3 2to3-${_pybasever} idle3 idle${_pybasever} pydoc3 pydoc${_pybasever}; do
+              sed -i "s|$(cygpath -w ${MINGW_PREFIX} | sed 's|\\|\\\\|g')/bin/python${_pybasever}.exe|/usr/bin/env python${_pybasever}.exe|g" "${pkgdir}${MINGW_PREFIX}"/bin/${fscripts}
+          done
+
+
+          cp "${pkgdir}${MINGW_PREFIX}"/bin/python3.exe "${pkgdir}${MINGW_PREFIX}"/bin/python.exe
+          cp "${pkgdir}${MINGW_PREFIX}"/bin/python3w.exe "${pkgdir}${MINGW_PREFIX}"/bin/pythonw.exe
+          cp "${pkgdir}${MINGW_PREFIX}"/bin/python3-config "${pkgdir}${MINGW_PREFIX}"/bin/python-config
+          cp "${pkgdir}${MINGW_PREFIX}"/bin/idle3 "${pkgdir}${MINGW_PREFIX}"/bin/idle
+          cp "${pkgdir}${MINGW_PREFIX}"/bin/pydoc3 "${pkgdir}${MINGW_PREFIX}"/bin/pydoc
```

Comment:
Do we want these here? In theory this is optional

Reply:
I think we would need the first 2 atleast.

Reply:
ok, then leave all for consistency